lib/good_job/active_job_extensions/logging.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+module GoodJob
+  module ActiveJobExtensions
+    module Logging
+      extend ActiveSupport::Concern
+
+      included do
+        around_perform do |job, block|
+          original_logger = job.logger
+          job.logger = ActiveSupport::TaggedLogging.new(ActiveSupport::Logger.new(LogDevice.new(job)).extend(ActiveSupport::Logger.broadcast(original_logger)))
+          block.call
+        ensure
+          job.logger = original_logger
+        end
+      end
+
+      class LogDevice
+        cattr_accessor :logs, default: Concurrent::Array.new
+
+        def initialize(job)
+          @job = job
+        end
+
+        def write(message)
+          self.class.logs << [@job.provider_job_id, message.strip]
+        end
+
+        def close
```

Comment:
I don't think #close and #reopen are overrode method but I might be wrong here. Are those methods mandatory or you think we'll be using them in the future ?

Reply:
I'm trying to match the interface of Ruby's `LogDevice`:

https://github.com/ruby/ruby/blob/9001e53e68d282493f513ed67824e4014fd01d57/lib/logger/log_device.rb

That said, I wonder if it would be better to inherit from LogDevice to be clear that I'm trying to match/override the interface, rather than just be an implicit duck-type.